include/ftxui/util/ref.hpp

```diff
@@ -59,19 +75,35 @@ class StringRef {
 /// class convert multiple immutable string toward a shared representation.
 class ConstStringRef {
  public:
-  ConstStringRef(const std::wstring* ref) : address_(ref) {}
-  ConstStringRef(std::wstring ref) : owned_(std::move(ref)) {}
+  ConstStringRef(const std::string* ref) : address_(ref) {}
+  ConstStringRef(const std::wstring* ref) : ConstStringRef(to_string(*ref)) {}
+  ConstStringRef(std::string ref) : owned_(std::move(ref)) {}
+  ConstStringRef(std::wstring ref) : ConstStringRef(to_string(ref)) {}
   ConstStringRef(const wchar_t* ref) : ConstStringRef(std::wstring(ref)) {}
   ConstStringRef(const char* ref)
       : ConstStringRef(to_wstring(std::string(ref))) {}
-  ConstStringRef(std::string ref)
-      : ConstStringRef(to_wstring(std::move(ref))) {}
-  const std::wstring& operator*() { return address_ ? *address_ : owned_; }
-  const std::wstring* operator->() { return address_ ? address_ : &owned_; }
+  const std::string& operator*() { return address_ ? *address_ : owned_; }
+  const std::string* operator->() { return address_ ? address_ : &owned_; }
+
+ private:
+  const std::string owned_;
+  const std::string* address_ = nullptr;
+};
+
+/// @brief An adapter. Reference a list of strings.
+class ConstStringListRef {
+ public:
+  ConstStringListRef(const std::vector<std::string>* ref) : ref_(ref) {}
+  ConstStringListRef(const std::vector<std::wstring>* ref) : ref_wide_(ref) {}
+
+  size_t size() const { return ref_ ? ref_->size() : ref_wide_->size(); }
+  std::string operator[](size_t i) const {
+    return ref_ ? (*ref_)[i] : to_string((*ref_wide_)[i]);
+  }
 
  private:
-  const std::wstring owned_;
-  const std::wstring* address_ = nullptr;
+  const std::vector<std::string>* const ref_ = nullptr;
+  const std::vector<std::wstring>* const ref_wide_ = nullptr;
```

Comment:
Consider removing the outer `const` from the data members.
```suggestion
  const std::vector<std::string>* ref_ = nullptr;
  const std::vector<std::wstring>* ref_wide_ = nullptr;
```

Ref: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-constref